crates/models/src/resources.rs

```diff
@@ -1,20 +1,25 @@
+use std::fmt::Display;
+use std::str::FromStr;
+
 use bytes::Bytes;
 use protocol::flow::ContentType as ProtoContentType;
 use schemars::JsonSchema;
 use serde::{Deserialize, Serialize};
-use serde_json::{from_value, json};
+use serde_with::{DeserializeFromStr, SerializeDisplay};
 
 use super::RelativeUrl;
 
+mod deserialization;
+mod serialization;
+
 /// A Resource is binary content with an associated ContentType.
-#[derive(Serialize, Deserialize, Debug, JsonSchema, Clone)]
+#[derive(Debug, JsonSchema, Clone)]
 #[serde(deny_unknown_fields, rename_all = "camelCase")]
 pub struct ResourceDef {
     /// # Content type of the Resource.
     pub content_type: ContentType,
-    /// # Byte content of the Resource.
-    #[serde(serialize_with = "as_base64", deserialize_with = "from_base64")]
-    #[schemars(schema_with = "base64_schema")]
+    /// # Content of the Resource.
+    #[schemars(schema_with = "serialization::content_schema")]
```

Comment:
Use an untagged enum for `content` which can be either a Bytes (or String if that's hard, and defer the base64-ing), OR an Object. Don't attempt to conjoin the two during ser/deser.

Instead, match on the type and do the conversion when we're inserting into the `models::tables::Resource`, _not_ `models::ResourceDef` while loading within the `sources` crate.

Prior art for this technique: `models/src/config.rs`, `models/src/schemas.rs`, etc.